index.js

```diff
@@ -47,49 +49,84 @@ class Rewt {
     this.options = options;
     this._redisConn = options.redisConn;
     this._scripty = new Scripty(this._redisConn);
+    this._shouldCacheSecret = options.cacheSecret || false;
   }
 
   /**
    * Signs the given payload.
    *
    * @param {Object} payload The payload to sign.
-   * @param {Function} cb Node style callback.
    */
-  sign(payload, cb) {
-    this._getSecret((err, val) => {
-      if (err) {
-        cb(err);
-      } else {
-        jwt.sign(payload, val, {}, cb);
-      }
+  async sign(payload) {
+    const secret = await this._getSecret();
+    return new Promise((resolve, reject) => {
+      jwt.sign(payload, secret, {}, (err, val) => {
+        if (err) {
+          this._clearCachedSecret();
+          return void reject(err);
+        }
+        return void resolve(val);
+      });
     });
   }
 
   /**
    * Verifies the given token and parses the payload from it.
    *
    * @param {String} token The token to verify and parse.
-   * @param {Function} cb Node style callback.
    */
-  verify(token, cb) {
-    this._getSecret((err, val) => {
-      if (err) {
-        cb(err);
-      } else {
-        jwt.verify(token, val, cb);
-      }
+  async verify(token) {
+    const secret = await this._getSecret();
+    return new Promise((resolve, reject) => {
+      jwt.verify(token, secret, (err, val) => {
+        if (err) {
+          this._clearCachedSecret();
```

Comment:
We should do a compare and clear here in case the cached secret was updated since we read it.

Reply:
Could you elaborate it a bit? My thinking was we'd just default to clearing the cached secret.

Reply:
Compare and clear -> check that the cached secret is the same as the local `secret` before clearing the cached secret, in case the cached secret expired and was updated by another `_getSecret` call after the local `secret` was read.